### PR TITLE
i18n(zh-cn): update `configuration-reference.mdx` & `internationalization.mdx`

### PR DESCRIPTION
**Pull request by liruifengv**
#### Description (required)

 update `configuration-reference.mdx` & `internationalization.mdx`

#### Related issues & labels (optional)

- https://github.com/withastro/docs/pull/9224
- https://github.com/withastro/docs/pull/9237
- https://github.com/withastro/docs/pull/9274

### CODE REVIEW

File: src/content/docs/zh-cn/reference/configuration-reference.mdx [hunk 1]
```diff
@@ -1186,6 +1186,41 @@ export default defineConfig({
...
+<p>
+
+**类型：**`"redirect" | "rewrite"`<br />
+**默认值：**`"redirect"`<br />
+<Since v="4.15.0" />
+</p>
+
+当 [`i18n.fallback`](#i18nfallback) 配置为避免为缺失的页面路由显示 404 页面时，此选项控制是否将用户[重定向](/zh-cn/guides/routing/#重定向)到回退页面，还是将回退页面的内容[重写](/zh-cn/guides/routing/#重写)到原始请求的 URL 上。
+
+默认情况下，Astro 的 i18n 路由会创建页面，将你的访问者重定向到基于你的回退配置的新目的地。浏览器将刷新并显示在 URL 栏中显示的目的地地址。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/reference/configuration-reference.mdx:1200]:
```suggestion
默认情况下，Astro 的 i18n 路由会创建页面，将你的访问者重定向到基于你的回退配置的新目的地。浏览器将刷新并在 URL 栏中显示目的地地址。
```

File: src/content/docs/zh-cn/reference/configuration-reference.mdx [hunk 2]
```diff
@@ -1186,6 +1186,41 @@ export default defineConfig({
...
+
+当 `i18n.routing.fallback: "rewrite"` 配置时，Astro 将创建页面，将回退页面的内容重写到原始请求的 URL 上。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/reference/configuration-reference.mdx:1202]:
```suggestion
当 `i18n.routing.fallback: "rewrite"` 配置时，Astro 则会创建页面，并将回退页面的内容重写到原始请求的 URL 上。
```

File: src/content/docs/zh-cn/reference/configuration-reference.mdx [hunk 3]
```diff
@@ -1186,6 +1186,41 @@ export default defineConfig({
...
+
+使用以下配置，如果你有 `src/pages/en/about.astro` 但不是 `src/pages/fr/about.astro`，`astro build` 命令将生成 `dist/fr/about.html` 与 `dist/en/about.html` 页面相同的内容。
+你的网站访问者将在 `https://example.com/fr/about/` 看到页面的英文版本  ，而不会被重定向。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/reference/configuration-reference.mdx:1205]:
```suggestion
使用以下配置，如果你有 `src/pages/en/about.astro` 这个文件，但没有 `src/pages/fr/about.astro` 时，`astro build` 命令将生成与 `dist/en/about.html` 页面相同内容的 `dist/fr/about.html` 页面。
你的网站访问者将在 `https://example.com/fr/about/` 看到页面的英文版本，而不会被重定向。
```

**Review by Nin3lee — CHANGES_REQUESTED** [review]

**Review by Nin3lee — APPROVED** [review]
LGTM!

**Review by yanthomasdev — APPROVED** [review]
LGTM! Good job! 🙌

**Pull request merged**